src/Sarif/Cache.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    ///  Cache is a simple LRU (least recently used) collection.
+    ///  The base class libraries don't offer one without an extra dependency.
+    /// </summary>
+    /// <typeparam name="TKey">Type of Keys in collection</typeparam>
+    /// <typeparam name="TValue">Type of Values being cached in collection</typeparam>
+    public class Cache<TKey, TValue> where TKey : IComparable<TKey>
+    {
+        private readonly Func<TKey, TValue> _builder;
+        private readonly Dictionary<TKey, TValue> _cache;
+        private readonly LinkedList<TKey> _keysInUseOrder;
+
+        public const int DefaultCapacity = 100;
+        public int Capacity { get; private set; }
+
+        /// <summary>
+        ///  Construct a new Cache
+        /// </summary>
+        /// <param name="builder">Method to build a value from a key when it isn't in the cache</param>
+        /// <param name="capacity">Maximum number of items to keep in cache, zero for no limit</param>
+        public Cache(Func<TKey, TValue> builder, int capacity = DefaultCapacity)
+        {
+            _builder = builder;
+            _cache = new Dictionary<TKey, TValue>();
+            _keysInUseOrder = new LinkedList<TKey>();
+            Capacity = capacity;
+        }
+
+        /// <summary>
+        ///  Returns the number of items currently cached
+        /// </summary>
+        public int Count => _cache.Count;
+
+        /// <summary>
+        ///  Returns the keys of items currently cached
+        /// </summary>
+        public IEnumerable<TKey> CachedKeys => _cache.Keys;
```

Comment:
IMO just `Keys` is a fine name.

Reply:
Yes, making it more consistent with Dictionary is a good idea.